src/displayapp/screens/StopWatch.cpp

```diff
@@ -123,96 +144,89 @@ void StopWatch::SetInterfaceStopped() {
-        lv_obj_realign(time);
-        isHoursLabelUpdated = true;
-      }
-    }
-    lv_label_set_text_fmt(msecTime, "%02d", currentTimeSeparated.hundredths);
-  } else if (currentState == States::Halted) {
-    const TickType_t currentTime = xTaskGetTickCount();
-    if (currentTime > blinkTime) {
-      blinkTime = currentTime + blinkInterval;
-      if (lv_obj_get_state(time, LV_LABEL_PART_MAIN) == LV_STATE_DEFAULT) {
-        lv_obj_set_state(time, LV_STATE_DISABLED);
-        lv_obj_set_state(msecTime, LV_STATE_DISABLED);
-      } else {
-        lv_obj_set_state(time, LV_STATE_DEFAULT);
-        lv_obj_set_state(msecTime, LV_STATE_DEFAULT);
-      }
+      lv_label_ins_text(lapText, LV_LABEL_POS_LAST, "\n");
     }
   }
 }
 
-void StopWatch::playPauseBtnEventHandler() {
-  if (currentState == States::Init || currentState == States::Halted) {
-    Start();
-  } else if (currentState == States::Running) {
-    Pause();
+void StopWatch::Refresh() {
+  if (stopWatchController.IsRunning()) {
+    RenderTime();
+  } else if (stopWatchController.IsPaused()) {
+    RenderPause();
+  }
+}
+
+void StopWatch::PlayPauseBtnEventHandler() {
+  if (stopWatchController.IsCleared() || stopWatchController.IsPaused()) {
+    stopWatchController.Start();
+    DisplayStarted();
+    wakeLock.Lock();
+  } else if (stopWatchController.IsRunning()) {
+    stopWatchController.Pause();
+    blinkTime = xTaskGetTickCount() + blinkInterval;
+    DisplayPaused();
+    RenderTime();
+    wakeLock.Release();
   }
 }
 
-void StopWatch::stopLapBtnEventHandler() {
-  // If running, then this button is used to save laps
-  if (currentState == States::Running) {
-    lv_label_set_text(lapText, "");
-    lapsDone = std::min(lapsDone + 1, maxLapCount);
-    for (int i = lapsDone - displayedLaps; i < lapsDone; i++) {
-      if (i < 0) {
-        lv_label_ins_text(lapText, LV_LABEL_POS_LAST, "\n");
-        continue;
-      }
-      TimeSeparated_t times = convertTicksToTimeSegments(laps[i]);
-      char buffer[17];
-      if (times.hours == 0) {
-        snprintf(buffer, sizeof(buffer), "#%2d    %2d:%02d.%02d\n", i + 1, times.mins, times.secs, times.hundredths);
-      } else {
-        snprintf(buffer, sizeof(buffer), "#%2d %2d:%02d:%02d.%02d\n", i + 1, times.hours, times.mins, times.secs, times.hundredths);
-      }
-      lv_label_ins_text(lapText, LV_LABEL_POS_LAST, buffer);
-    }
-  } else if (currentState == States::Halted) {
-    Reset();
+void StopWatch::StopLapBtnEventHandler() {
+  if (stopWatchController.IsRunning()) {
+    stopWatchController.AddLapToHistory();
+    RenderLaps();
+  } else if (stopWatchController.IsPaused()) {
+    stopWatchController.Clear();
+    DisplayCleared();
+    wakeLock.Release();
   }
 }
 
 bool StopWatch::OnButtonPushed() {
-  if (currentState == States::Running) {
-    Pause();
+  if (stopWatchController.IsRunning()) {
+    stopWatchController.Pause();
+    DisplayPaused();
+    wakeLock.Release();
```

Comment:
Is RenderTime needed here (like the button handler)?

Reply:
Actually more broadly is there any different for the pause button handler and side button handler to be different? Maybe they could be unified

Reply:
In InfiniTime 1.14, the on-screen button is a play/pause toggle while the side button is not: when pressed in the `Paused` state, it exits the screen. This PR preserves that behavior.

I do find the side button slightly counter-intuitive and occasionally exit the screen without meaning to. Having both buttons work the same way would improve UI consistency.

Having said that, I just noticed that pausing with the side button is much slower than with the on-screen button (like 300-400 vs. 150 ms) so the consitency is debatable. And, of course, some people may be used to exiting via the side button; I can't estimate how much of an issue that is.

On the balance, I'm in favor of keeping existing side button behavior. You are right, however, that the transition to the `Paused` state should be the same in both cases so I've extracted it into a common method.

Reply:
Yeah the real kicker with the side button is that it triggers on button release rather than press I think - improving this is definitely out of scope for this PR though as you say